portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java

```diff
@@ -2938,6 +2962,137 @@ private DLVersionNumberIncrease _computeDLVersionNumberIncrease(
 			previousDLFileVersion, nextDLFileVersion);
 	}
 
+	private void _expireFileEntriesByCompanyId(
+			Date expirationDate, Map<String, Serializable> workflowContext,
+			ServiceContext serviceContext, Long companyId)
+		throws PortalException {
+
+		long userId = _getActiveCompanyAdminUserId(companyId);
+
+		List<DLFileEntry> fileEntries =
+			_getFileEntriesByCompanyIdAndExpirationDate(
+				companyId, expirationDate);
+
+		for (DLFileEntry fileEntry : fileEntries) {
+			DLFileVersion latestFileVersion =
+				dlFileVersionLocalService.fetchLatestFileVersion(
+					fileEntry.getFileEntryId(), false);
+
+			if (latestFileVersion.isExpired()) {
+				continue;
+			}
+
+			if (_log.isDebugEnabled()) {
+				_log.debug(
+					StringBundler.concat(
+						"Expiring file entry ", fileEntry.getFileEntryId(),
+						" with expiration date ",
+						fileEntry.getExpirationDate()));
+			}
+
+			updateStatus(
+				userId, fileEntry, latestFileVersion,
+				WorkflowConstants.STATUS_EXPIRED, serviceContext,
+				workflowContext);
+
+			_notifySubscribers(
+				userId, latestFileVersion, _buildEntryURL(latestFileVersion),
+				"expire", new ServiceContext());
+
+			_notifyOwner(
+				userId, latestFileVersion, _buildEntryURL(latestFileVersion),
+				"expire", new ServiceContext());
+		}
+	}
+
+	private long _getActiveCompanyAdminUserId(long companyId)
+		throws PortalException {
+
+		Role role = roleLocalService.getRole(
+			companyId, RoleConstants.ADMINISTRATOR);
+
+		long[] userIds = userLocalService.getRoleUserIds(role.getRoleId());
+
+		User user = null;
+
+		if (!ArrayUtil.isEmpty(userIds)) {
+			for (long userId : userIds) {
```

Comment:
This chunk of code is duplicated a couple of times. Can you extract it into a private method?